<a href="https://colab.research.google.com/github/MDTulum/NetGEN/blob/main/Question3_Assignment1_Reg.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Question 3

part a

In [1]:
hw1_data = read.csv("https://raw.githubusercontent.com/hgweon2/ss3859/master/hw1_data1.csv")
idx = which(hw1_data$x1 < 4)
a = length(idx)

cat("# of obs where x1 lower than 4: ", a)


# of obs where x1 lower than 4:  40

part b

In [2]:
idx = which((hw1_data$x1 < 4) & (hw1_data$x2 == "L"))
b = length(idx)
cat("# of obs where x1 lower than 4 & x2 is L: ", b)


# of obs where x1 lower than 4 & x2 is L:  32

part c

In [3]:
idx = which(hw1_data$x2 == "L")
A = hw1_data[idx,]
cat("Mean:", mean(A$x1), "\n")
cat("median:", median(A$x1), "\n")
cat("sd:", sd(A$x1), "\n")

Mean: 2.581517 
median: 2.567377 
sd: 2.152698 


part d

H0 = 4 and H1 not equal to 4.

In [4]:
x = hw1_data$x1
sample_mean = mean(x)
cat("Mean of X1 (x_bar): ", sample_mean, "\n")
sample_sd = sd(x)
n = length(x)
t_stat = (sample_mean - 4)/(sample_sd/sqrt(n))
p_val = 2*(1 - pt(abs(t_stat),df=n-1))
cat("p_val: ", p_val,"\n")
# we can do the same test using function t.test
t.test(x = x, mu = 4, alternative = c("two.sided"), conf.level = 0.95)

Mean of X1 (x_bar):  4.434816 
p_val:  0.08871225 



	One Sample t-test

data:  x
t = 1.7192, df = 99, p-value = 0.08871
alternative hypothesis: true mean is not equal to 4
95 percent confidence interval:
 3.932958 4.936674
sample estimates:
mean of x 
 4.434816 


Since the p-value is 8.87%, which is more than significance level of 5%, then there is not enough evidence that the true mean of x1 differs from 4, so we fail to reject H0 of mean being 4.

part e

the same test as d but we use the subset A (from part c) of only those with x2=L and test against 2 instead of 4. the other change is that the hypothesis is about being greater than something so no two sided distribution any more and only one side (right side for greater) is considered

In [5]:
idx = which(hw1_data$x2 == "L")
A = hw1_data[idx,]
x = A$x1
sample_mean = mean(x)
cat("Mean of X1 (x_bar): ", sample_mean, "\n")
sample_sd = sd(x)
n = length(x)
t_stat = (sample_mean - 2)/(sample_sd/sqrt(n))
p_val = 1 - pt(t_stat,df=n-1)
cat("p_val: ", p_val,"\n")
# we can do the same test using function t.test
t.test(x = x, mu = 2, alternative = c("greater"), conf.level = 0.95)

Mean of X1 (x_bar):  2.581517 
p_val:  0.04187624 



	One Sample t-test

data:  x
t = 1.7714, df = 42, p-value = 0.04188
alternative hypothesis: true mean is greater than 2
95 percent confidence interval:
 2.02936     Inf
sample estimates:
mean of x 
 2.581517 


Since the p-value is 4.2%, which is less than the cut off of 5%, the statement that the true mean of x1 where x2 = L is greater than 2 is convincing.

